specifyweb/specify/migrations/0004_schema_config_update.py

```diff
@@ -0,0 +1,177 @@
+"""
+This migration updates the Schema Config entries for pre-geo tables. 
+Geo migration added new fields to already existing tables. This migration updates the Schema Config to reflect those changes.
+
+Fields added:
+Collection -> collectionObjectType
+GeographyTreeDef -> discipline
+GeologicTimePeriodTreeDef -> discipline
+LithostratTreeDef -> discipline
+StorageTreeDef -> institution
+TaxonTreeDef -> discipline
+"""
+from django.db import migrations
+
+FIELD_DATA = [
+    {
+        "table": "Collection",
+        "field": "collectionObjectType",
+        "isrequired": False,
+    },
+    {
+        "table": "GeographyTreeDef",
+        "field": "discipline",
+        "isrequired": True,
+    },
+    {
+        "table": "GeologicTimePeriodTreeDef",
+        "field": "discipline",
+        "isrequired": True,
+    },
```

Comment:
If we want to be able to create Discipline's anytime soon and keep compatibility with Specify 6, we can't mark `GeographyTreeDef -> discipline` or `GeologicTimePeriod -> discipline` as required unfortunately. 

The inverse relationships, `Discipline -> geographyTreeDef` and `Discipline -> geologicTimePeriodTreeDef` are required at the database level. 

We should actually modify the datamodel file to reflect this as well (and regenerate the frontend datamodel and testing files): 
https://github.com/specify/specify7/blob/f429d81de93184126bf42bd91ee8e6e2166cd23f/specifyweb/specify/datamodel.py#L3845
https://github.com/specify/specify7/blob/f429d81de93184126bf42bd91ee8e6e2166cd23f/specifyweb/specify/datamodel.py#L3961

https://github.com/specify/specify7/blob/f429d81de93184126bf42bd91ee8e6e2166cd23f/specifyweb/frontend/js_src/lib/components/DataModel/types.ts#L9-L17
https://github.com/specify/specify7/blob/production/specifyweb/frontend/js_src/lib/tests/ajax/static/context/datamodel.json

Reply:
What about LithostratTreeDef -> discipline and TaxonTreeDef -> discipline?

Reply:
If we want, we can leave `LithostratTreeDef -> discipline` and `TaxonTreeDef -> discipline` as required at the datamodel level (as long as it's not required at the database level). 
The `Discipline -> taxonTreeDef` and `Discipline -> lithoStratTreeDef` relationships are not required. 

```sql
MariaDB [ojsmnh]> desc discipline;
+-----------------------------+-------------+------+-----+---------+-------+
| Field                       | Type        | Null | Key | Default | Extra |
+-----------------------------+-------------+------+-----+---------+-------+
| UserGroupScopeId            | int(11)     | NO   | PRI | NULL    |       |
| TimestampCreated            | datetime    | NO   |     | NULL    |       |
| TimestampModified           | datetime    | YES  |     | NULL    |       |
| Version                     | int(11)     | YES  |     | NULL    |       |
| ModifiedByAgentID           | int(11)     | YES  | MUL | NULL    |       |
| CreatedByAgentID            | int(11)     | YES  | MUL | NULL    |       |
| disciplineId                | int(11)     | YES  |     | NULL    |       |
| Name                        | varchar(64) | YES  | MUL | NULL    |       |
| RegNumber                   | varchar(24) | YES  |     | NULL    |       |
| Type                        | varchar(64) | YES  |     | NULL    |       |
| TaxonTreeDefID              | int(11)     | YES  | MUL | NULL    |       |
| DataTypeID                  | int(11)     | NO   | MUL | NULL    |       |
| GeologicTimePeriodTreeDefID | int(11)     | NO   | MUL | NULL    |       |
| GeographyTreeDefID          | int(11)     | NO   | MUL | NULL    |       |
| DivisionID                  | int(11)     | NO   | MUL | NULL    |       |
| LithoStratTreeDefID         | int(11)     | YES  | MUL | NULL    |       |
| IsPaleoContextEmbedded      | bit(1)      | YES  |     | NULL    |       |
| PaleoContextChildTable      | varchar(50) | YES  |     | NULL    |       |
| TectonicUnitTreeDefID       | int(11)     | YES  | MUL | NULL    |       |
+-----------------------------+-------------+------+-----+---------+-------+
19 rows in set (0.022 sec)
```

Just for completeness, `Institution -> storageTreeDef` is also not required.